confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -18,133 +18,258 @@
 
 import json
 import logging
+import sys
+import urllib
+from copy import deepcopy
+from threading import Lock
 
-from requests import Session
+from requests import Session, utils
 
-from .error import SchemaRegistryClientError
-from .schema import SchemaProvider
+from .schema import RegisteredSchema, Schema
+from ..serialization.error import SchemaRegistryError
 
-log = logging.getLogger(__name__)
+# TODO: use six or adopt compat file practicelike requests.
+if sys.version < '3':
+    def _urlencode(value):
+        return urllib.quote(value, safe='')
+else:
+    def _urlencode(value):
+        return urllib.parse.quote(value, safe='')
+
+VALID_AUTH_PROVIDERS = ['URL', 'USER_INFO', 'SASL_INHERIT']
 
-"""
-API Endpoint templates
-"""
-SCHEMAS = 'schemas/ids/{}?format=serialized'
-SUBJECT = 'subjects/{}'
-SUBJECTS = 'subjects'
-VERSION = 'subjects/{}/versions/{}'
-VERSIONS = 'subjects/{}/versions'
-CONFIG = 'config/{}'
-CONFIGS = 'config'
+log = logging.getLogger(__name__)
 
 
-def url_formatter(template, *args):
+class _RestClient(object):
     """
-    Constructs a url string from a template.
+    Manages SchemaRegistryClient HTTP communication
 
     Args:
-        template: url format string
-        args: args to use when resolving url string
-
+        src_conf (dict): Dictionary containing _RestClient configuration
     """
-    return template.format(*args)
+    default_headers = {'Accept': "application/vnd.schemaregistry.v1+json,"
+                                 " application/vnd.schemaregistry+json,"
+                                 " application/json"}
 
+    def __init__(self, src_conf):
+        self.session = Session()
+        conf = self.handle_conf(src_conf)
+        self.base_url = conf['url']
 
-class SchemaRegistryClient(object):
-    """
-    A client that talks to a Schema Registry over HTTP
+    def __del__(self):
+        self.session.close()
 
-    See Also:
-        http://confluent.io/docs/current/schema-registry/docs/intro.html
+    def handle_conf(self, src_conf):
```

Comment:
this method is public - doesn't seem right.

i wouldn't have bothered making another method for this.